docs/docs/tutorials/orca-yarn-tutorial.md

```diff
@@ -0,0 +1,532 @@
+    ```bash
+    export ARROW_LIBHDFS_DIR=/opt/cloudera/parcels/CDH-${CHD_VERSION}/lib64/
+    ```
+
+# 3. Prepare Dataset 
+To run the example on YARN, you should upload the Fashion-MNIST dataset to a distributed storage (such as HDFS or S3).   
+
+First, please download the Fashion-MNIST dataset manually on your __Client Node__ (where you submit the program to YARN).
+```bash
+# PyTorch official dataset download link
+git clone https://github.com/zalandoresearch/fashion-mnist.git
+
+mv /path/to/fashion-mnist/data/fashion /path/to/local/data/FashionMNIST/raw 
+```
+Then upload it to a distributed storage.
+```bash
+# Upload to HDFS
+hdfs dfs -put /path/to/local/data/FashionMNIST hdfs://url:port/path/to/remote/data
+```
+
+# 4. Prepare Custom Modules
+Spark allows to upload Python files (`.py`), and zipped Python packages (`.zip`) to the executors by setting `--py-files` option in Spark scripts or `extra_python_lib` in `init_orca_context`. 
+
+The FasionMNIST example needs to import modules from `model.py`.
+* When using `python` command, please specify `extra_python_lib` in `init_orca_context`.
+    ```python
+    from bigdl.orca import init_orca_context, stop_orca_context
+    from model import model_creator, optimizer_creator
+
+    # Please switch the `cluster_mode` to `yarn-cluster` when running on cluster mode.
+    init_orca_context(cluster_mode="yarn-client", cores=4, memory="10g", num_nodes=2,
+                      driver_cores=2, driver_memory="4g",
+                      extra_python_lib="model.py")
+    ```
+
+    Please see more details in [Orca Document](https://bigdl.readthedocs.io/en/latest/doc/Orca/Overview/orca-context.html#python-dependencies).
+
+* When using `bigdl-submit` or `spark-submit` script, please specify `--py-files` option in the script.
+    ```bash
+    --py-files model.py
+    ```
+
+    Import custom modules at the beginning of the example:
+    ```python
+    from bigdl.orca import init_orca_context, stop_orca_context
+    from model import model_creator, optimizer_creator
+
+    init_orca_context(cluster_mode="bigdl-submit") # or spark-submit
+    ```
+
+    Please see more details in [Spark Document](https://spark.apache.org/docs/latest/submitting-applications.html). 
+
+__Note:__
+* You could follow the steps below to use a zipped package instead (recommended if your program depends on a nested directory of Python files) :
+    1. Compress the directory into a zipped package.
+        ```bash
+        zip -q -r FashionMNIST_zipped.zip FashionMNIST
+        ```
+    2. Please upload the zipped package (`FashionMNIST_zipped.zip`) to YARN.
+        * When using `python` command, please specify `extra_python_lib` argument in `init_orca_context`.
+      
+        * When using `bigdl-submit` or `spark-submit` script, please specify `--py-files` option in the script.
+    3. You should import custom modules from the unzipped file as below.
```

Comment:
You can then import the custom modules from the unzipped file in your program as below.